compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/OperatorPrecedence.java

```diff
@@ -43,11 +43,12 @@
 
     ANON_FUNC_OR_LET(16),   //  (x) => y
     QUERY(17),              //  from x, select x, where x
+    TRAP(18),               // (trap x)
```

Comment:
If anything is not breaking, can use the same level as query, for the trap. 
```suggestion
    QUERY_OR_TRAP(17),      //  from x, select x, where x, (trap x)
```

If we are gonna have a new trap level, let's add an entry in the following two places just to be safe.
https://github.com/ballerina-platform/ballerina-lang/blob/42eeaedbdb4756e09095b5da350839ef8825efeb/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java#L3066
https://github.com/ballerina-platform/ballerina-lang/blob/42eeaedbdb4756e09095b5da350839ef8825efeb/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java#L3112

Reply:
+1 will update

Reply:
Merging all 3 `ANON_FUNC_OR_LET`, `QUERY`, `TRAP` broke ` infer-anonymous-function-expr` parsing.
Merging `QUERY` and `TRAP` broke parsing some remote method invocations (https://scans.gradle.com/s/ogebtsi4kd2g4)
Hence had to lower the `TRAP` below `ACTION`.